Core/Python/add_device_to_static_group.py

```diff
@@ -93,11 +242,22 @@ def add_device_to_static_group(ome_ip_address, ome_username, ome_password, group
     PARSER = argparse.ArgumentParser(description=__doc__,
                                      formatter_class=RawTextHelpFormatter)
     PARSER.add_argument("--ip", "-i", required=True, help="OME Appliance IP")
-    PARSER.add_argument("--user", "-u", required=True,
-                        help="Username for OME Appliance", default="admin")
+    PARSER.add_argument("--user", "-u", required=False,
+                        help="Username for the OME Appliance", default="admin")
     PARSER.add_argument("--password", "-p", required=True,
-                        help="Password for OME Appliance")
+                        help="Password for the OME Appliance")
     PARSER.add_argument("--groupname", "-g", required=True,
-                        help="A valid name for the group")
+                        help="The name of the group to which you want to add servers.")
+    exclusive_group = PARSER.add_mutually_exclusive_group(required=True)
+    exclusive_group.add_argument("--devicenames", "-n", help="The names of the device you want to add to the group in "
+                                                             "format: \'device1,device2,device3,etc\'")
+    exclusive_group.add_argument("--devicetags", "-t", help="A list of service tags which you want to add to the group "
+                                                            "in format: \'tag1,tag2,tag3,etc\'")
     ARGS = PARSER.parse_args()
-    add_device_to_static_group(ARGS.ome_ip_address, ARGS.user, ARGS.password, ARGS.groupname)
+
+    if ARGS.devicetags:
```

Comment:
Not affecting functionality - just personal preference and not a reason to change the code

Would the required=True param for the mutually exclusive group guarantee that one of these is present? I'd think it does - in that case there may not be a need for an explicit else check for the devicenames check (elif ARGS.devicenames) - it would simply fall to the else case where you can assume ARGS.devicenames is valid

if ARGS.devicetags:
  do something
else:
  do something else

Reply:
hahaha that's fair. This is probably just in line with my comment about starting in C. I always lean toward more strict checking rather than less unless I'm in a scenario where I'm writing an application with a high loop count and I know every line counts. In this case I did the else if because it guarantees me that the entry parameters will always be as I expect them even if someone else changes some code around.

I can change it if you'd like me to.